dev/lint-scala

```diff
@@ -17,6 +17,8 @@
 # limitations under the License.
 #
 
+set -e
+
```

Comment:
This is a bit of a weird patch. The mvn command should actually continue checking for `$?` and not for the error output being empty. In particular because it parses for "Unformatted files"

Not the second issue is that the lint-scala should not care about the code in 


```
resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/ExecutorKubernetesCredentialsFeatureStepSuite.scala
```

The mvn command explicitly just checks for 

```
    -pl sql/api \
    -pl sql/connect/common \
    -pl sql/connect/server \
    -pl connector/connect/client/jvm \
```